### PR TITLE
HDDS-11609. Switch to Recon v2 UI as the default UI

### PR DESCRIPTION
**Pull request by devabhishekpal**
## What changes were proposed in this pull request?
HDDS-11609. Switch to Recon v2 UI as the default UI

Please describe your PR in detail:
* Currently the default UI on visiting Recon is the V1 UI
* This PR changes the default UI to show the new v2 UI by default and enable the UI switching for users that want to use the old UI
* The heatmap page for this points to the v1 Heatmap UI even when v2 UI is enabled which would be changed to point to new heatmap page once that is prepared.
* As a part of this PR we are also adding a new improved 404 page design.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-11609

## How was this patch tested?
Patch was tested manually

#### New UI is the default UI (Switch is not toggled)
Screenshot 2024-11-05 at 14 04 36

----

#### Old UI (Switch is toggled)
Screenshot 2024-11-05 at 14 04 53

----

#### New 404 page
Screenshot 2024-10-25 at 22 18 20

### CODE REVIEW

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.tsx [hunk 1]
```diff
@@ -56,48 +57,51 @@ class App extends React.Component<Record<string, object>, IAppState> {
...
           }
           <Layout className={layoutClass}>
             <Header>
               <div style={{ margin: '16px 0', display: 'flex', justifyContent: 'space-between' }}>
-                {(enableNewUI) ? <BreadcrumbsV2 /> : <Breadcrumbs />}
+                {(enableOldUI) ? <Breadcrumbs /> : <BreadcrumbsV2 />}
                 <AntDSwitch
-                  disabled={true}
-                  checkedChildren={<div style={{ paddingLeft: '2px' }}>New UI</div>}
+                  unCheckedChildren={<div style={{ paddingRight: '2px' }}>UI v2</div>}
```
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.tsx:77]: Its better to rename labels shown to users as old and new UI as "Switch to Old UI" and "Switch to New UI"

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.tsx [hunk 2]
```diff
@@ -32,22 +32,23 @@ import classNames from 'classnames';
 import Loader from '@/v2/components/loader/loader';
 
 import './app.less';
+import NotFound from '@/v2/pages/notFound/notFound';
```
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.tsx:35]: So this is new page for `notFound 404`, but if it is imported here, how it is being used across Recon for all APIs in generic way ?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.tsx (hunk 1 above)
Comment (devabhishekpal) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.tsx]: Addressed in [9e942d4](https://github.com/apache/ozone/pull/7358/commits/9e942d4368083a49aecc038b9498b18793008f11)
After change:
Screenshot 2024-10-29 at 12 15 33

**Review by devmadhuu — COMMENTED** [review]
Thanks @devabhishekpal for improving the patch. All LGTM +1. Just a nit: Update your screenshot in PR which reflects updated toggle switch.

**Comment by devabhishekpal** [comment]
Thanks @devmadhuu. Updated the screenshots in description to reflect the new changes.

**Review by ArafatKhan2198 — APPROVED** [review]
Thanks for the patch @devabhishekpal changes LGTM +1

**Pull request merged**